pkg/ali/components/Networking.vue

```diff
@@ -0,0 +1,701 @@
+      const out = [{ value: '', label: this.t('ack.networking.resourceGroupId.default') }, ...this.allResourceGroups];
+
+      return out;
+    },
+    vpcOptions() {
+      return Object.keys(this.allVPCs).map((vpcId) => {
+        return { value: vpcId, label: this.allVPCs[vpcId].label || '' };
+      });
+    },
+    availabilityZoneOptions() {
+      return this.allAvailabilityZones;
+    },
+    podVswitchIdOptions() {
+      const unformatted = this.vswitchIds || [];
+
+      return unformatted.map((vswitchId) => {
+        return { value: vswitchId, label: this.allVSwitches[vswitchId]?.label || '' };
+      });
+    },
+    vswitchIdOptions() {
+      return Object.keys(this.allVSwitches).map((vswitchId) => {
+        return { value: vswitchId, label: this.allVSwitches[vswitchId]?.label || '' };
+      });
+    },
+  },
+
+  watch: {
+    chooseVPC(val) {
+      if (this.isNewOrUnprovisioned) {
+        if (val) {
+          this.$emit('update:zoneIds', []);
+          this.$emit('update:vpcId', this.vpcOptions[0]?.value || '');
+        } else {
+          this.$emit('update:vpcId', '');
+          this.$emit('update:vswitchIds', []);
+          this.$emit('update:podVswitchIds', []);
+          if (this.allAvailabilityZones.length) {
+            this.$emit('update:zoneIds', this.allAvailabilityZones.map((z) => z.value));
+          }
+        }
+      }
+    },
+    fvFormIsValid: {
+      immediate: true,
+      handler(neu) {
+        this.$emit('update:configIsValid', neu);
+      }
+    },
+    fvUnreportedValidationErrors: {
+      immediate: true,
+      handler(neu) {
+        this.$emit('update:configUnreportedErrors', neu);
+      }
+    },
+
+    'config.regionId': {
+      handler(neu, old) {
+        if (neu && neu !== old) {
+          this.$emit('update:resourceGroupId', '');
+          this.$emit('update:vpcId', '');
+          this.$emit('update:vswitchIds', []);
+          this.$emit('update:podVswitchIds', []);
+          this.$emit('update:zoneIds', []);
+
+          this.getResourceGroups();
+          this.getZones();
+        }
+      },
+      immediate: true
+    },
+    'config.alibabaCredentialSecret': {
+      handler(neu, old) {
+        if (neu && neu !== old) {
+          this.$emit('update:resourceGroupId', '');
+          this.$emit('update:vpcId', '');
+          this.$emit('update:vswitchIds', []);
+          this.$emit('update:podVswitchIds', []);
+          this.$emit('update:zoneIds', []);
+
+          this.getResourceGroups();
+          this.getZones();
+        }
+      },
+      immediate: true
+    },
+    resourceGroupId: {
+      async handler(neu, old) {
+        if (neu !== old) {
+          this.$emit('update:vpcId', '');
+          this.$emit('update:vswitchIds', []);
+          this.$emit('update:podVswitchIds', []);
+
+          await this.getVPCs();
+          const firstVpc = Object.keys(this.allVPCs)[0];
+
+          if (firstVpc) {
+            this.$emit('update:vpcId', firstVpc);
```

Comment:
I think this is running into the same issue as the `vpcId` watcher, where it runs when the component loads, setting `vpcId` even though `enableVPC` is `false`. It looks like you could update this to:

`if (firstVpc && this.enableVPC) {`